### PR TITLE
provide helper script to build Caddy with nginx adapter

### PR DESCRIPTION
**Pull request by mohammed90**
There have been 2 reports (#5 and https://caddy.community/t/howto-use-nginx-config-adapter/6929) of difficulties to build Caddy with the nginx adapter. This PR provides a helper script to do just that.

### CODE REVIEW

File: build_caddy.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+mkdir caddy || true
```
Comment (mholt) [inline build_caddy.sh]: Could maybe be `mkdir -p caddy`?

File: build_caddy.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+cd caddy || true
+curl "https://raw.githubusercontent.com/caddyserver/caddy/v2/cmd/caddy/main.go" > main.go
+sed -i.bak 's/^)/	_ "github.com\/caddyserver\/nginx-adapter"\'$'\n)/' main.go && rm main.go.bak
+
+go mod init caddy
+grep 'require' ../go.mod >> go.mod
```
Comment (mholt) [inline build_caddy.sh]: Not sure I follow this one

**Review by mholt — COMMENTED** [review]
Cool, great start! As you said, will need some instructions. I just have a couple questions but nothing stands out to me as incorrect/flawed.

File: build_caddy.sh (hunk 2 above)
Comment (mohammed90) [reply build_caddy.sh]: The go command keeps requiring beta9, so I'm just copying the `require` line to depend on the same version the adapter depends on. This go.mod file is created on-demand, so we can't fix it to a certain version. If you remember, we had similar issue in my initial PR.

Comment (mholt) [reply build_caddy.sh]: `go mod edit` might be a better, more proper way to do that, maybe? https://golang.org/cmd/go/#hdr-Edit_go_mod_from_tools_or_scripts

Or even `go get` if all we're doing is setting "require"

Comment (mohammed90) [reply build_caddy.sh]: Ok, used `go mod edit`. I used `go list` to read out the currently required version of Caddy so we don't need to change it in many places on upgrades. Looks better?

**Review by mholt — APPROVED** [review]
Awesome, looks great! Thank you!!

**Pull request merged**